test/unit/location_expansion/location_expansion_test.bzl

```diff
@@ -0,0 +1,54 @@
+"""Unittest to verify location expansion in rustc flags"""
+
+load("@bazel_skylib//lib:unittest.bzl", "analysistest")
+load("//rust:defs.bzl", "rust_library")
+load("//test/unit:common.bzl", "assert_action_mnemonic", "assert_argv_contains")
+
+def _location_expansion_rustc_flags_test(ctx):
+    env = analysistest.begin(ctx)
+    tut = analysistest.target_under_test(env)
+    action = tut.actions[0]
+    argv = action.argv
+    assert_action_mnemonic(env, action, "Rustc")
+    assert_argv_contains(env, action, "test/unit/location_expansion/mylibrary.rs")
+    expected = "@${pwd}/" + ctx.bin_dir.path + "/test/unit/location_expansion/generated_flag.data"
+    assert_argv_contains(env, action, expected)
+    return analysistest.end(env)
+
+location_expansion_rustc_flags_test = analysistest.make(_location_expansion_rustc_flags_test)
+
+def _location_expansion_test():
+    native.genrule(
+        name = "flag_generator",
+        outs = ["generated_flag.data"],
+        cmd = "echo --cfg=test_flag > $@",
+    )
+
+    rust_library(
+        name = "mylibrary",
+        srcs = ["mylibrary.rs"],
+        rustc_flags = [
+            "@$(location :flag_generator)",
```

Comment:
Nit: you could marginally simplify the test by not passing a file with cfg flags, but by passing a cfg flag with value (https://doc.rust-lang.org/rustc/command-line-arguments.html#--cfg-configure-the-compilation-environment).

Something like (hope this works, haven't tried):

```
rustc_flags = [ "--cfg=", "yololo=$(location :foo)"],
```

Reply:
I leave the decision what looks simpler to you.

Reply:
This won't work because the analysis fails if the target isn't a dependency of the rule (in compile_data here).